homeassistant/components/shelly/light.py

```diff
@@ -369,3 +408,27 @@ def _update_callback(self) -> None:
         self.control_result = None
         self.mode_result = None
         super()._update_callback()
+
+
+class RpcShellyLight(ShellyRpcEntity, LightEntity):
+    """Entity that controls a light on RPC based Shelly devices."""
+
+    def __init__(self, wrapper: RpcDeviceWrapper, key: str) -> None:
+        """Initialize light."""
+        super().__init__(wrapper, key)
+        self._id = key[7:]
+
+    @property
+    def is_on(self) -> bool:
+        """If light is on."""
+        return bool(self.wrapper.device.status[self.key]["output"])
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on light."""
+        await self.set_state("Switch.Set", {"id": self._id, "on": True})
```

Comment:
We should rename set_state to call_rpc or something. This is confusing.

Reply:
Renamed to `call_rpc`